torch/csrc/jit/passes/onnx/scalar_type_analysis.cpp

```diff
@@ -110,10 +110,10 @@ static c10::optional<c10::ScalarType> InferExpectedScalarType(const Node* n) {
         } else if (nkind == onnx::Constant) {
           typesFromScalars.emplace_back(
               input->node()->t(attr::value).scalar_type());
-        } else if (
-            auto scalar_type =
-                input->type()->cast<TensorType>()->scalarType()) {
-          typesFromTensors.emplace_back(*scalar_type);
+        } else if (auto tensor_type = input->type()->cast<TensorType>()) {
+          auto scalar_type = tensor_type->scalarType();
```

Comment:
Do we need a check similar to the below check on input_tensor_type, and what happens in cases with no scalar_type

Reply:
this pass is only for the inputs that have scalar type information recorded, so I don't think there is anything to be done if there is no scalar_type